enhancements/unmanaged-clair-database.md

```diff
@@ -0,0 +1,142 @@
+
+There are two steps required for this implementation:
+
+- A user has the ability to provide a custom clair config as a secret reference in the CRD
+- A clairdatabase component should exist in the CRD
+
+## Motivation
+
+There are several scenarios in which a user of the Quay Operator might want to use an unmanaged database. For example, to allow for the operator to work in a Geo-Replicated environment, multiple instances of the operator must be able to talk to the same database. In the current implementation, Clair can only talk to its managed database inside of a single cluster. Additionally, a user might have a requirement for an HA clair database that exists outside of a cluster.
+
+### Goals
+
+- A `clairdatabase` component should be configurable in the Operator CR.
+- An unmanaged `clairdatabase` component should check for the existence of a custom clair configuration file
+- The Operator should validate the custom configuration provided when the component is set to unmanaged (the config must fit the clair schema, and the database connection must be working)
+- When `clairdatabase` is marked as unmanaged, any `clair-postgres` pods should be stopped and or not provisioned at all
+
+### Non-Goals
+
+- The setup of an unmanaged Clair database is the responsibility of the user and is out of the scope of the operator
+
+## Proposal
+
+The Operator should allow for a user to provide their own Clair configuration in the following manner.
+
+In the provided `spec.configBundleSecret`, an additional field called `clair-config.yaml` should be accepted. An example secret might look like the following:
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: config-bundle-secret
+  namespace: quay-enterprise
+data:
+  config.yaml: <base64 encoded Quay config>
+  clair-config.yaml: <base64 encoded Clair config>
+```
+
+Once the secret has been created, a user should also be able to unmange the `clairdatabase` component. An example CR might look like the following:
+
+```
+apiVersion: quay.redhat.com/v1
+kind: QuayRegistry
+metadata:
+  name: quay
+spec:
+  components:
+  - kind: clairdatabase
+    managed: false
+  configBundleSecret: config-bundle-secret
+
+```
+
+The provided clair-config.yaml would be mounted into the Clair pod, and any fields not provided should be automatically populated with defaults using the Clair config module.
+
+### Implementation Details/Notes/Constraints
+
+One implementation detail to note is that a managed Clair component should consume a custom `clair-config.yaml` as long as it is provided. This is similar to how the base Quay component can be customized with a `config.yaml` despite always being a managed component. However, the clair config must be in sync with the state of the `clairdatabase` component. For example, if a user chooses a managed `clairdatabase` component, the `clair-config.yaml` should not contain custom DSN values.
+
+### Risks and Mitigations
```

Comment:
There is one extra concern that I haven't seen you mention in this document: do we need to change anything in the config-tool to support this new feature?

Reply:
The new config module comes with some schema validation. We would need to validate the database connection in addition to this. This could be done using the config-tool module that exists today. See this function here -> https://github.com/quay/config-tool/blob/171819fa5bca7b3a55be146766abd453dcab43b1/pkg/lib/shared/validators.go#L458. We will likely need to change it around a bit to be more generic, but the validation logic can be reused.